### PR TITLE
[SE-0115] Rename *LiteralConvertible protocols to ExpressibleBy*…

### PR DESCRIPTION
**Pull request by CodaFi**
#### What's in this pull request?

A slew of renamings for [SE-0115](https://github.com/apple/swift-evolution/blob/master/proposals/0115-literal-syntax-protocols.md).  Patches for [swift-corelibs-foundation](https://github.com/apple/swift-corelibs-foundation/pull/447), [swiftpm](https://github.com/apple/swift-package-manager/pull/484), and [swift-corelibs-xctest](https://github.com/apple/swift-corelibs-xctest/pull/136) have been prepared.  The existing protocols are now deprecated typealiases that will be hard errors in a future patch.
#### Resolved bug number: ([SE-0115](https://github.com/apple/swift-evolution/blob/master/proposals/0115-literal-syntax-protocols.md))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci please test.

**Comment by gribozavr** [comment]
:shipit:

**Comment by CodaFi** [comment]
@swift-ci please test and merge.

File: stdlib/public/core/CompilerProtocols.swift [hunk 1]
```diff
@@ -708,3 +708,67 @@ public protocol _DestructorSafeContainer {
...
 public typealias BooleanType = Boolean
+
+// Deprecated by SE-0115.
+
+@available(*, deprecated, renamed: "ExpressibleByNilLiteral")
+public typealias NilLiteralConvertible
+  = ExpressibleByNilLiteral
+@available(*, deprecated, renamed: "_ExpressibleByBuiltinIntegerLiteral")
+public typealias _BuiltinIntegerLiteralConvertible
+  = _ExpressibleByBuiltinIntegerLiteral
```
Comment (rintaro) [inline stdlib/public/core/CompilerProtocols.swift]: @gribozavr Just a question. Do we *need* migration support for underscored things?

Comment (gribozavr) [inline stdlib/public/core/CompilerProtocols.swift]: No, we don't need it.

**Comment by CodaFi** [comment]
OS X test failure is unrelated and was just patched by Doug.

File: stdlib/public/core/CompilerProtocols.swift (hunk 1 above)
Comment (CodaFi) [inline stdlib/public/core/CompilerProtocols.swift]: I wasn't sure who was and wasn't using them.  They will be gone in the revision that properly deprecates them.

**Pull request merged**